packages/typegpu/src/core/texture/texture.ts

```diff
@@ -360,6 +335,29 @@ class TgpuTextureImpl implements TgpuTexture, INTERNAL_TgpuTexture {
     return new TgpuFixedStorageTextureImpl(params ?? {}, access, this);
   }
 
+  createView(
+    access: 'mutable' | 'readonly' | 'writeonly' | 'sampled',
+    params?: TextureViewParams<GPUTextureViewDimension, GPUTextureFormat>,
+  ) {
+    if (access === 'sampled') {
+      return this.asSampled(params);
+    }
+
+    const storageParams = params as TextureViewParams<
+      StorageTextureDimension,
+      StorageTextureTexelFormat
+    >;
+
+    switch (access) {
+      case 'mutable':
+        return this.asMutable(storageParams);
+      case 'readonly':
+        return this.asReadonly(storageParams);
+      case 'writeonly':
+        return this.asWriteonly(storageParams);
```

Comment:
maybe these `.as*` methods could have a leading underscore to denote their internal purpose

Reply:
+1